### PR TITLE
WooCommerce Debug helper

### PR DESCRIPTION
**Pull request by abhaymaniyar**

### CODE REVIEW

File: includes/helpers/simpl_logger.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+function is_debug_mode_enabled()
+{
+    return (
+        'yes' == get_option('wc_settings_tab_simpl_debug_logs')
+    );
+}
+
+function simpl_log($level, $message)
+{
+    if (is_debug_mode_enabled()) {
```
Comment (piyush-getsimpl) [inline includes/helpers/simpl_logger.php:146]: Should we keep alert, critical, error, emergency outside of settings control to always log ?

File: includes/hook_handlers/order-hook-handler.php [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <?php
 function order_hook($order_id)
 {
+    $logger = new SimplLogger();
```
Comment (piyush-getsimpl) [inline includes/hook_handlers/order-hook-handler.php:4]: Do we really want to instantiate everytime or make SimplLogger singleton ?

File: includes/helpers/simpl_logger.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+ * 'error': Error conditions.
+ * 'warning': Warning conditions.
+ * 'notice': Normal but significant condition.
+ * 'info': Informational messages.
+ * 'debug': Debug-level messages.
+ * @param string $message Message to log.
+ */
+define('SIMPL_LOG_NAME', 'simpl-logs');
+
+class SimplLogger {
```
Comment (piyush-getsimpl) [inline includes/helpers/simpl_logger.php:19]: Nomenclature - Simpl_Logger

File: includes/helpers/simpl_logger.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+{
+    return (
+        'yes' == get_option('wc_settings_tab_simpl_debug_logs')
+    );
+}
+
+function simpl_log($level, $message)
+{
+    if (is_debug_mode_enabled()) {
+        $logger = wc_get_logger();
```
Comment (piyush-getsimpl) [inline includes/helpers/simpl_logger.php:147]: Do we need to invoke this everytime? Can we add this at the class level itself ?

File: simpl-checkout-woocommerce.php [hunk 5]
```diff
@@ -71,6 +71,15 @@ function simpl_sentry_exception($err) {
...
 
+$simpl_logger = null;
+function get_simpl_logger() {
+    if (!isset($simpl_logger)) {
+       $simpl_logger = new Simpl_Logger();
+    }
+
+    return $simpl_logger;
+}
+
```
Comment (Chhekur) [inline simpl-checkout-woocommerce.php:82]: this get method can be part of same logger class itself as static method.
so we can initiate the logger here once and then we start using get method to get the logger instance, and logger class itself can hold the singleton logic.

**Review by Chhekur — APPROVED** [review]

**Pull request merged**